### DISCUSSION TITLE
Bar at the bottom right or left corner of screen

### DISCUSSION THREAD

**Discussion by dancaer69** [Alexays/Waybar#2208] [Q&A] [answered]
Is there any way to achive this? I have a bar with only tray module with a specified width/height. The bar appears at the bottom as I want but is in center. How can I move it to right corner?

**Comment by zefr0x** [comment]
Don't specify width, just use the margin option:

option | typeof | default | description
-- | -- | -- | --
margin | string |   | Margins value using the css format without units.
margin-<top\|left\|bottom\|right> | integer |   | Margins value without units.

**Reply by dancaer69** [reply]
I already tried this. When set it to left or right, is vertical and at the top side of the screen. I want to be horizontal at the bottom and right side of screen.

**Reply by zefr0x** [reply]
If you want your bar to be vertical ( | ) at the bottom right, than specify `"position": "right"` and **don't specify** `height` and set `margin-top`.

If you want your bar to be horizontal ( _ ) at the bottom right, than specify `"position": "bottom"` and **don't specify** `width` and set `margin-left`.

I'm not sure if I am understanding what you mean, but this works with me.

If this isn't what you meant, maybe do it using `style.css` by setting the `background-color` of the bar to `transparent` and moving things using margin and padding.

**Reply by dancaer69** [reply]
I want the bar to be horizontal at bottom right. If I don't specify width, then bar takes all bottom screen area and I don't want this. If set a margin-left(I suppose you mean in css file), is ignored.

**Reply by zefr0x** [reply, marked as answer]
No, I meant the `margin-left` option in the `config` file, **not** in the `style.css` file. It will work as the `width` option, but from a single side.

Remember to not specify the `width` option because it doesn't work with the `margin` option in the `config` file.

**Reply by dancaer69** [reply]
Okay, I was trying with css option(at list most times). So, yes this works and I finally the bar moved to the left without to occupy all space at bottom. But seems that there is a bug, because I still cannot move it and resize it the way I want and after some poind it crashes the WM. To be more specific: Iuse eww widgets as a sidebar which has width 240 and it's at the right side of screen. I use waybar only for taskbar and tray, I what I want to achive is to have it also at the right side, at bottom and with width 240 also, so to be below the eww sidebar. To achive this I need to set a left margin 1680(my screen's width is 1920), but when I input a value above 1040, the WM crashes.

**Comment by dancaer69** [comment]
Okay, I was doing it from css file. This way works, but seems that there is a bug though. If I use a margin above 1280 the Window Manager crashes with segmentation fault. I tried to use only right side modules in case that was the problem(I had the taskbar left and the tray right), but it doesn't make any difference. Above this value the WM crashes and I want a bigger value for bar to have a width 240, so need a value of 1680.
**UPDATE:**
I found why was crashing. I have a dual monitor setup and was caused because the second monitor has width 1280. So by specify the output in config file I can use a bigger value.
